src/profile/components/multiItemAddressRow/MultiItemAddressRow.tsx

```diff
@@ -0,0 +1,189 @@
+import { AddressValue } from '../../helpers/editData';
+import { createFormFieldHelpers } from '../../helpers/formik';
+import { addressSchema } from '../../../common/schemas/schemas';
+import FormButtons from '../formButtons/FormButtons';
+import EditButtons from '../editButtons/EditButtons';
+import FormikDropdown, {
+  HdsOptionType,
+} from '../../../common/formikDropdown/FormikDropdown';
+import getLanguageCode from '../../../common/helpers/getLanguageCode';
+import LabeledValue from '../../../common/labeledValue/LabeledValue';
+import getCountry from '../../helpers/getCountry';
+import { getFormFields } from '../../helpers/formProperties';
+import SaveIndicator from '../saveIndicator/SaveIndicator';
+import { useCommonEditHandling } from '../../hooks/useCommonEditHandling';
+import { RowItemProps } from '../multiItemEditor/MultiItemEditor';
+
+type FormikValues = AddressValue;
+
+function MultiItemAddressRow(props: RowItemProps): React.ReactElement {
+  const { data, testId, dataType } = props;
+  const value = data.value as AddressValue;
+  const { address, city, postalCode, countryCode } = value;
+  const { t, i18n } = useTranslation();
+  const lang = i18n.languages[0];
+  const applicationLanguage = getLanguageCode(i18n.languages[0]);
+  const countryList = countries.getNames(applicationLanguage);
+  const countryOptions = Object.keys(countryList).map(key => ({
+    value: key,
+    label: countryList[key],
+  }));
+  const formFields = getFormFields(dataType);
+  const {
+    isNew,
+    isEditing,
+    actionHandler,
+    currentAction,
+  } = useCommonEditHandling(props);
+  const { hasFieldError, getFieldErrorMessage } = createFormFieldHelpers<
+    FormikValues
+  >(t, isNew);
+
+  const { primary } = data;
+
+  if (isEditing) {
+    return (
+      <Formik
+        initialValues={{
+          address,
+          city,
+          postalCode,
+          countryCode,
+        }}
+        onSubmit={async values => {
+          await actionHandler('save', values);
+        }}
+        validationSchema={addressSchema}
+      >
+        {(formikProps: FormikProps<FormikValues>) => (
+          <Form className={commonFormStyles.multiItemForm}>
+            <h4 className={commonFormStyles.sectionTitle}>
+              {primary
+                ? t('profileInformation.primaryAddress')
+                : t('profileInformation.address')}
+            </h4>
+            <div className={commonFormStyles.multiItemWrapper}>
+              <Field
+                name="address"
+                id={`${testId}-address`}
+                maxLength={formFields.address.max as number}
+                as={TextInput}
+                invalid={hasFieldError(formikProps, 'address')}
+                aria-invalid={hasFieldError(formikProps, 'address')}
+                helperText={getFieldErrorMessage(formikProps, 'address')}
+                labelText={t(formFields.address.translationKey)}
```

Comment:
The font-weight and size of these input labels seem to change depending on mode (edit or view). It might be what is wanted, but at first glimpse looked little weird.

Reply:
Looks like HDS and layouts have different styles. HDS should be followed. Another fix to be done in the HDS update....